### PR TITLE
SQL Migration Extension Skeleton

### PR DESCRIPTION
**Pull request by aaomidi**

### CODE REVIEW

File: extensions/sql-migration/package.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
+{
+  "name": "sql-migration",
+  "displayName": "%displayName%",
+  "description": "%description%",
+  "version": "0.0.0",
```
Comment (Charles-Gagnon) [inline extensions/sql-migration/package.json:5]: Suggest just starting at 0.0.1 so we don't have to update it later

File: extensions/sql-migration/src/controllers/mainController.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { Controller } from '../models/controller';
+
+export class MainController implements Controller {
```
Comment (Charles-Gagnon) [inline extensions/sql-migration/src/controllers/mainController.ts:9]: Do we really need to do this controller stuff? It never really seems worth the extra layer of abstraction

File: extensions/sql-migration/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+		"module": "commonjs",
+		"outDir": "./out",
+		"lib": [
+			"es6", "es2015.promise"
+		],
+		"emitDecoratorMetadata": true,
+		"experimentalDecorators": true,
+		"moduleResolution": "node",
+		"declaration": false,
+		"strict": false,
```
Comment (Charles-Gagnon) [inline extensions/sql-migration/tsconfig.json:14]: Not enabling strict?

File: extensions/sql-migration/package.json [hunk 4]
```diff
@@ -0,0 +1,41 @@
+{
+  "name": "sql-migration",
+  "displayName": "%displayName%",
```
Comment (Charles-Gagnon) [inline extensions/sql-migration/package.json:3]: Missing loc file

File: extensions/sql-migration/src/controllers/mainController.ts (hunk 2 above)
Comment (aaomidi) [reply extensions/sql-migration/src/controllers/mainController.ts:9]: That's true, I guess we don't really need it. It kinda makes dealing with disposables easier but yeah I'll try to think of a cleaner way of doing this.

File: extensions/sql-migration/tsconfig.json (hunk 3 above)
Comment (aaomidi) [reply extensions/sql-migration/tsconfig.json:14]: Good idea, let me see if I can add some extra rules for this project too.

**Pull request merged**